### PR TITLE
code-sum of digits of a number

### PR DESCRIPTION
**Pull request by sarayusreeyadavpadala**
I have changed the code. This code is beginner friendly and more modular. Let me know if any improvements are needed.

### CODE REVIEW

**Review by OfficialAhmed — CHANGES_REQUESTED** [review]
Double check your code please

File: Sum of digits of a number.py [hunk 1]
```diff
@@ -1,7 +1,31 @@
...
+
+def addition(num):
+    Sum=0
+    if type(num) is type(None):
+        print("Try again!")
+        sys.exit()
+    while num > 0:
+        digit = int(num % 10)
+        Sum = Sum + digit
+        num = num / 10
```
Comment (OfficialAhmed) [inline Sum of digits of a number.py:24]: no big deal `num /= 10` instead of `num = num / 10`

Comment (OfficialAhmed) [reply Sum of digits of a number.py:24]: again not important but makes the code clean `Sum += digit`

**Review by OfficialAhmed — APPROVED** [review]
No issues great work but cleaning few lines might make it a bit more user friendly

File: Sum of digits of a number.py [hunk 2]
```diff
@@ -1,7 +1,31 @@
...
+
+def get_integer():
+    for i in range(3,0,-1):
+        num = input("enter a number:")
+        if num.isnumeric():
+            num = int(num)
+            return num
+        else:
+            print("enter integer only")
+            print(f'{i-1} chances are left')
```
Comment (NitkarshChourasia) [inline Sum of digits of a number.py:12]: Use {chance if i <=1 else chances} use something like this, to have a proper string literal output for "chances" and "chance."

File: Sum of digits of a number.py (hunk 1 above)
Comment (NitkarshChourasia) [reply Sum of digits of a number.py:24]: Indeed, readability is also a good coding goal to aim for.

File: Sum of digits of a number.py [hunk 3]
```diff
@@ -1,7 +1,31 @@
...
+    while num > 0:
+        digit = int(num % 10)
+        Sum = Sum + digit
+        num = num / 10
+    return Sum
+       
+
+number = get_integer()
+Sum = addition(number)
+print(f'sum of digits of {number} is {Sum}')
```
Comment (NitkarshChourasia) [inline Sum of digits of a number.py:30]: Document the code further with,
- Docstrings
- Comments

Good, for others to read and understand your code.

**Review by NitkarshChourasia — CHANGES_REQUESTED** [review]

File: Sum of digits of a number.py (hunk 3 above)
Comment (NitkarshChourasia) [reply Sum of digits of a number.py:30]: Also if possible include unittest in it, you know.

Comment (NitkarshChourasia) [inline Sum of digits of a number.py:30]: also use,
if __name__ == "__main__":
    function1()
    function2()
    function3()
    
    
    
    
    
    
    
 # Something adding to the bottom of the file is good, practice.

**Review by NitkarshChourasia — CHANGES_REQUESTED** [review]

File: Sum of digits of a number.py (hunk 3 above)
Comment (sarayusreeyadavpadala) [reply Sum of digits of a number.py:30]: > also use, if **name** == "**main**": function1() function2() function3()
> 
> # Something adding to the bottom of the file is good, practice.

Could you explain this a little bit more @NitkarshChourasia.

File: Sum of digits of a number.py (hunk 1 above)
Comment (sarayusreeyadavpadala) [reply Sum of digits of a number.py:24]: > again not important but makes the code clean `Sum += digit`

Yes. I'll change this.

File: Sum of digits of a number.py (hunk 2 above)
Comment (sarayusreeyadavpadala) [reply Sum of digits of a number.py:12]: For sure. Thank u @NitkarshChourasia

**Comment by sarayusreeyadavpadala** [comment]
> No issues great work but cleaning few lines might make it a bit more user friendly
 Thank u @OfficialAhmed

File: Sum of digits of a number.py (hunk 3 above)
Comment (NitkarshChourasia) [reply Sum of digits of a number.py:30]: Try chatGPT or google.com, but do implement it.

Comment (NitkarshChourasia) [reply Sum of digits of a number.py:30]: @sarayusreeyadavpadala https://www.tutorialspoint.com/What-does-the-if-name-main-do-in-Python

Comment (sarayusreeyadavpadala) [reply Sum of digits of a number.py:30]: > Try chatGPT or google.com, but do implement it.

Okay @NitkarshChourasia

**Pull request closed without merging**